lib/srv/regular/sshserver.go

```diff
@@ -1064,6 +1079,23 @@ func (s *Server) HandleNewChan(ctx context.Context, ccx *sshutils.ConnectionCont
 	}
 }
 
+func (s *Server) canPortForward(scx *srv.ServerContext, channel ssh.Channel) bool {
+	// Is the node configured to allow port forwarding?
+	if !s.allowTCPForwarding {
+		writeStderr(channel, "Node does not allow port forwarding")
+		return false
+	}
+
+	// Check if the role allows port forwarding for this user.
+	err := s.authHandlers.CheckPortForward(scx.DstAddr, scx)
+	if err != nil {
+		writeStderr(channel, err.Error())
+		return false
+	}
+
+	return true
+}
+
```

Comment:
Return an error with the error message here and then use `writeStderr` in the caller. Lowercase `Node`.